File: 1512 number-of-good-pairs/solution.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        int arr[] = new int[101];
        for(int num:nums) arr[num]++;
        int sum=0;
        for(int i:arr){
            if(i>1) sum+=(i*(i-1))/2;
        }
        return sum;
    }
}

File: 1512 number-of-good-pairs/solution.go
func numIdenticalPairs(nums []int) int {
    ma := make([]int,101)
    for _,v:=range nums {
        ma[v]++
    }
    sum:=0
    for _,v := range ma {
        if v>1 {
            sum+=v*(v-1)/2
        }
    }
    return sum
}
